File: tests/fixtures/listeners/handlers/typescript-handlers/src/add.ts
/**
 * TypeScript handler for Calculator.Add operation.
 * Strongly typed: takes AddRequest returns AddResponse
 */

import type { AddRequest, AddResponse } from "./types.ts";

/**
 * Add two numbers.
 *
 * @param request - AddRequest with 'a' and 'b' as int32 values
 * @returns AddResponse with 'result' field containing the sum
 */
export function handler(request: AddRequest): AddResponse {
  // Perform calculation using strongly-typed inputs
  const result: number = request.a + request.b;

  // Create and return strongly-typed response
  const response: AddResponse = { result };
  return response;
}

File: examples/python-openapi-listener/calculator/add.py
"""
Python handler for Calculator.Add operation.
Strongly typed: takes AddRequest returns AddResponse
"""
from calculator.types import AddRequest, AddResponse


def handler(request: AddRequest) -> AddResponse:
    """
    Add two numbers.

    Args:
        request: AddRequest with 'a' and 'b' as int32 values

    Returns:
        AddResponse with 'result' field containing the sum
    """
    # Perform calculation using strongly-typed inputs
    result: int = request["a"] + request["b"]

    # Create and return strongly-typed response
    response: AddResponse = {"result": result}
    return response
